doc/threading_model.md

```diff
@@ -0,0 +1,31 @@
+﻿# gRPC Threading Model

```

Comment:
changed the title. Regarding the file name, I suppose I can do that when the file gets submitted to grpc.io ?